### PR TITLE
Move HBONE section into its own doc

### PR DESCRIPTION
**Pull request by bleggett**
## Description

We've had a glossary description that sort of handwaves at what HBONE is, but given how often we use the acronym and the fact that it's a nontrivial part of the ambient security model, it probably needs its own Architecture section that can be referenced from other spots.

This mostly just moves the existing HBONE section out of the ztunnel architecture doc into its own, smaller, (hopefully) clearer thing.

Fixes: https://github.com/istio/istio.io/issues/13482

## Reviewers

- [x] Ambient
- [x] Docs
- [ ] Installation
- [x] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure
- [ ] Localization/Translation

### CODE REVIEW

File: content/en/docs/ambient/architecture/hbone/index.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
+title: HBONE
```
Comment (louiscryan) [inline content/en/docs/ambient/architecture/hbone/index.md:2]: Consider renaming to "TLS & Tunneling" and define HBONE within the content here.

When we link to this doc from elsewhere we will just mention TLS

File: content/en/docs/ambient/architecture/hbone/index.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+description: Understanding Istio's secure tunneling protocol.
+weight: 2
+owner: istio/wg-networking-maintainers
+test: no
+---
+
+
+## Understanding the HBONE protocol
+
+HBONE (HTTP Based Overlay Network Encapsulation) is an Istio-specific term. It is a mechanism to transparently multiplex packets related to many different application connections over a single, mTLS encrypted network connection - an encrypted tunnel.
```
Comment (louiscryan) [inline content/en/docs/ambient/architecture/hbone/index.md:12]: 'packets' or 'TCP streams' ?

similarly in the "... many different application connections..." section. 

No need to pressage QUIC :)

Comment (bleggett) [reply content/en/docs/ambient/architecture/hbone/index.md:12]: yeah `packets` is left over from the previous doc. Agree tho.

**Comment by bleggett** [comment]
DNM-ing until https://github.com/istio/istio.io/pull/14994 goes in

**Review by louiscryan — APPROVED** [review]
LGTM

**Review by MorrisLaw — APPROVED** [review]

**Comment by craigbox** [comment]
Both are now in.

Don't we use HBONE between sidecars?

**Comment by bleggett** [comment]
> Both are now in.
> 
> Don't we use HBONE between sidecars?

Yep, resynced this against those two, should be good to go.

**Comment by bleggett** [comment]
@craigbox note that this also does https://github.com/istio/istio.io/issues/15005, PTAL

**Comment by bleggett** [comment]
oops, forgot this would merge even if I added @craigbox directly as a reviewer. if he doesn't want some/all of this it I'll back it out.

**Comment by craigbox** [comment]
i'm fine with treating master as a dev branch until we release!  Merge early and often please

**Pull request merged**